router/src/server.rs

```diff
@@ -452,23 +453,39 @@ async fn generate_stream(
     Sse<impl Stream<Item = Result<Event, Infallible>>>,
 ) {
     let span = tracing::Span::current();
-    let on_message_callback = |stream_token: StreamResponse| {
-        let event = Event::default();
-        event.json_data(stream_token).unwrap()
-    };
     let (headers, response_stream) =
-        generate_stream_internal(infer, compute_type, Json(req), on_message_callback, span).await;
-    let sse = Sse::new(response_stream).keep_alive(KeepAlive::default());
+        generate_stream_internal(infer, compute_type, Json(req), span).await;
+
+    let final_response_stream = async_stream::stream! {
+        let mut response_stream = Box::pin(response_stream);
+        while let Some(raw_event) = response_stream.next().await {
+            match raw_event {
+                Ok(stream_token) => {
+                    let event = Event::default();
+                    let event = event.json_data(stream_token).unwrap();
+                    yield Ok(event);
+                }
+                Err(_err) => {
+                     let event = Event::default();
+                    yield Ok(event);
```

Comment:
Why ? You're changing functionality here.

Reply:
great catch, Thanks. I've updated the logic to emit an event in both cases of a token or an error (to match the existing functionality).